### PR TITLE
[IA-1610] RStudio shutdown script not running reliably

### PR DESCRIPTION
**Pull request by rtitle**
The real fix is some systemd changes to make our startup/shutdown scripts dependent on `docker.service`.

But since I was updating the Dataproc image anyway, I updated all the docker image versions and added `terra-jupyter-gatk`, `terra-rstudio-base`, and `terra-rstudio-bioconductor`.

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've documented my API changes in Swagger

In all cases:

- [ ] Get a thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: jenkins/Jenkinsfile
Comment (rtitle) [inline jenkins/Jenkinsfile]: Revert before merge

File: jenkins/Jenkinsfile
Comment (rtitle) [inline jenkins/Jenkinsfile]: Revert before merge

File: jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh
Comment (rtitle) [inline jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh:169]: The actual fix for the RStudio shutdown issue

**Comment by rtitle** [comment]
Custom dataproc build with these changes: https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-dataproc-image/234/

Testing Leo in a fiab with the generated dataproc images.

**Comment by rtitle** [comment]
Whoops the dataproc job failed with `no space left on device`. May need to bump the image size.

File: jenkins/Jenkinsfile
Comment (Qi77Qi) [inline jenkins/Jenkinsfile:167]: I'm thinking maybe we should make a map of tool -> customImage, and have separate image for RStudio vs jupyter to save some space...

This map will be useful nice we start to support multiple versions of spark versions

Comment (rtitle) [reply jenkins/Jenkinsfile:167]: Yeah that might make sense going forward, there are a lot of images here..

**Comment by rtitle** [comment]
New build, it takes a while now: https://fc-jenkins.dsp-techops.broadinstitute.org/job/leonardo-dataproc-image/236/

File: jenkins/Jenkinsfile
Comment (rtitle) [reply jenkins/Jenkinsfile:167]: I feel like I'd want to do that in a separate PR though. I could also leave out the anvil-rstudio images for now..

**Comment by rtitle** [comment]
I successfully generated dataproc images using this branch and tested them using a Leo fiab. It seems to work! All the expected images are cached on the machine:
```
REPOSITORY                                                  TAG                 IMAGE ID            CREATED             SIZE
broadinstitute/terra-jupyter-gatk                           0.0.8               55e4825caf8d        6 hours ago         5.32GB
us.gcr.io/broad-dsp-gcr-public/terra-jupyter-gatk           0.0.8               55e4825caf8d        6 hours ago         5.32GB
us.gcr.io/broad-dsp-gcr-public/terra-jupyter-hail           0.0.5               f73d54d71136        7 days ago          5.44GB
us.gcr.io/broad-dsp-gcr-public/terra-jupyter-python         0.0.6               03b7198630e3        7 days ago          4.67GB
us.gcr.io/broad-dsp-gcr-public/terra-jupyter-bioconductor   0.0.9               2742e8b364da        3 weeks ago         4.52GB
us.gcr.io/broad-dsp-gcr-public/terra-jupyter-r              0.0.7               b5f374b2a028        3 weeks ago         2.89GB
us.gcr.io/broad-dsp-gcr-public/terra-jupyter-base           0.0.6               cf52c97a772e        4 weeks ago         2.35GB
us.gcr.io/anvil-gcr-public/anvil-rstudio-bioconductor       0.0.3               ffaa06c46e4e        4 weeks ago         4.32GB
us.gcr.io/anvil-gcr-public/anvil-rstudio-base               0.0.2               a816f564af5b        5 weeks ago         2.5GB
us.gcr.io/broad-dsp-gcr-public/welder-server                60e28bc             eef2c7e01353        2 months ago        1.81GB
us.gcr.io/broad-dsp-gcr-public/leonardo-jupyter             5c51ce6935da        4ed49e25e0ba        4 months ago        6.39GB
broadinstitute/openidc-proxy                                2.3.1_2             b1514b13020b        2 years ago         311MB
```

I also verified the RStudio fix, and did some notebooks smoke testing.

So I think this is good to merge, pending thumb (@Qi77Qi @jdcanas). Once this is merged I'll kick off a new dataproc image, and then merge the subsequent generated Leo PR.

File: jenkins/Jenkinsfile
Comment (jdcanas) [reply jenkins/Jenkinsfile:167]: We could also theoretically generalize the terra-docker job/conf to have a repo/dir specification for each image, or include the images as a submodule to allow them to be built and scanned like the other images if we want to have more ownership

This would allow that job to update the reference.conf above and cascade updates to the custom dataproc image. Might be worth a follow-up ticket to cut down on future manual updating. As of now, an update to the reference.conf will cause the new images to be built into the custom image though, so it may not be super necessary

**Review by jdcanas — APPROVED** [review]
lgtm

**Pull request merged**
